core/internal/runhistoryreader/parquet/iterator/parquet_data_iterator.go

```diff
@@ -56,17 +59,41 @@ func NewParquetDataIterator(
 
 		selectedRows:    selectedRows,
 		selectedColumns: selectedColumns,
+		fileReader:      reader,
 
 		recordReader: recordReader,
 		current:      current,
+		lastStep:     0,
 	}, nil
 }
 
+func (r *ParquetDataIterator) UpdateQueryRange(
+	minValue float64,
+	maxValue float64,
+	selectAll bool,
+) error {
+
+	// Update the selected range
+	r.selectedRows.minValue = minValue
+	r.selectedRows.maxValue = maxValue
```

Comment:
`minValue` and `maxValue` are private properties of `SelectedRowsRange` and shouldn't be updated here. I don't think that struct should be mutated; maybe you could pass a new `SelectedRowsRange` to this function.

Is an UpdateQueryRange function really necessary? I don't immediately see what the advantage is over simply closing the previous iterator and making a new one. This seems overcomplicated.